File: Assets/UnityMobileModules/Bluetooth/Bluetooth.cs
namespace UnityMobileModules
{
	/// <summary>
	/// Bluetooth information
	/// </summary>
    public static partial class Bluetooth
    {
        /// <summary>
        /// Does the device support Bluetooth
        /// </summary>
        public static bool deviceSupportsBluetooth
        {
            get
            {
#if UNITY_EDITOR
                return false;
#elif UNITY_ANDROID
                return DeviceSupportsBluetooth_Android;
#elif UNITY_IOS

#endif
            }
        }

        /// <summary>
        /// Is bluetooth enabled
        /// </summary>
        public static bool isEnabled
        {
            get
            {
                if (!deviceSupportsBluetooth) return false;
#if UNITY_EDITOR
                return false;
#elif UNITY_ANDROID
                return IsEnabled_Android;
#elif UNITY_IOS

#endif
            }
        }
    }
}

File: Assets/Android Studio Project~/UnityMobileModules/src/main/java/com/UnityMobileModules/Bluetooth.java
package com.UnityMobileModules;

import android.bluetooth.BluetoothAdapter;

public class Bluetooth {

    /**
     * Does the device support Bluetooth
     * @return - If the device supports Bluetooth
     */
    public static boolean DeviceSupportsBluetooth(){
        BluetoothAdapter mBluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
        return mBluetoothAdapter != null;
    }

    /**
     * Is Bluetooth enabled?
     * @return - If Bluetooth is enabled
     */
    public static boolean IsEnabled(){
        BluetoothAdapter mBluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
        //Requires android.permission.BLUETOOTH - Safe
        return mBluetoothAdapter.isEnabled();
    }

//    /**
//     * Are there any devices connected
//     * @return - All connected devices
//     */
//    public static String[] GetDevicesConnected(){
//        BluetoothAdapter mBluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
//        Set<BluetoothDevice> pairedDevices = mBluetoothAdapter.getBondedDevices();
//
//        //List to return
//        ArrayList<String> devices = new ArrayList<>();
//
//        if (pairedDevices.size() > 0) {
//            for (BluetoothDevice device : pairedDevices) {
//                String deviceString = device.getName() + "|||" + device.getBluetoothClass().getMajorDeviceClass() + "|||" + device.getBluetoothClass().getDeviceClass();
//                devices.add(deviceString);
//            }
//        }
//
//        //Converts ArrayList to String[]
//        return devices.toArray(new String[0]);
//    }
}
